syntax = "proto3";

enum SelectOption {
    GET = 0;
    GET_ALL = 1;
    ADD = 2;
    REGISTER=3;
    LOGIN=4;
    CURRENT_USER=5;
    REFRESH_TOKEN=6;
    CREATE = 7;
    UPDATE =8;
    REDUCE = 9;
    DELETE =10;
    PAYMENT_DONE = 11;
}


message Inventory {
    int32 id =1;
    string product_id=2;
    string inventory_id =3;
    int64 stock_level = 4;
    int64 add_stock_level = 5;
    int64 reduce_stock_level = 6;
    int64 reserved_stock = 7;
    string name = 8;
    string description =9;
    float price =10;
    bool is_available = 11;
    string error_message = 12;
    SelectOption option =13;
    int64 http_status_code = 14;
}

message Product {
    int32 id =1;
    string product_id=2;
    string name = 3;
    string description =4;
    float price =5;
    bool is_available = 6;
    string message = 7;
    string error_message = 8;
    SelectOption option =9;
    int64 http_status_code = 10;
}

enum OrderStatus {
  IN_PROGRESS = 0;
  COMPLETED = 1;
}
enum PaymentStatus {
    PENDING = 0;
    PAID = 1;
    FAILED = 2;
}

message Order {
    int32 id =1;
    string order_id=2;
    string product_id =3;
    string user_id = 4;
    int64 quantity= 5;
    string shipping_address = 6;
    string customer_notes = 7;
    int64 stock_level = 8;
    bool is_product_available = 9;
    bool is_stock_available  = 10;
    string message = 11;
    string error_message = 12;
    OrderStatus order_status = 13;
    PaymentStatus payment_status = 14;
    SelectOption option =15;
    int64 http_status_code = 16;
}

message InventoryList {
    repeated Inventory inventories = 1;
}